### ISSUE TITLE
Add early stopping to trainer(s)

### ISSUE THREAD

**Issue by tkornut** [IBM/pytorchpipe#69]
Generally it is already incorporated (as best_* checkpoint contain only models from moment in time when validation loss was smallest), but despite that the trainer continues until one of the other terminal conditions will occur.

Idea:
  - monitor "total loss" (add it to trainer statistics)
  - store "minimal_loss" value and "loss_down_counter" (indicating when last time loss went down) 
  - introduce config variable "early_stop" (number of episodes/epochs to stop when loss is not going down) (DEFAULT: 10?)
